### DISCUSSION TITLE
Not able to extend DatabaseManager

### DISCUSSION THREAD

**Discussion by nick-nds** [laravel/framework#46855] [Q&A]
I need to extend DatabaseManager to register a new connection as stated in API documentation, https://laravel.com/api/10.x/Illuminate/Database/DatabaseManager.html#method_extend

I am using code:

```
app('db')->extend($connectioName, function ($config, $name) use ($databaseName) {
  Log::info("set database connecttion");
  $config['database'] = $databaseName;
});
```

I am not able to set the Database Name also I am not able to see any Log message. I have checked using different ways but I am sure, the closure defined is not being executed. I am not able to identify the problem. Need help.

**Comment by ttrig** [comment]
Do you really want to both extend `DatabaseManager` **and** register a new connection?

Just add a connection in `config/database.php` if you want a new connection?

`$connectioName` should probably be `$connectionName`.

**Reply by nick-nds** [reply]
Yes, I thought so. Adding a new connection in `config/database.php` for a new connection. But there are some scenarios where we need to register a new connection as a fallback. 
The `$connectioName` is a typo. In the code, it is spelled correctly.

However, my issue is resolved. There was some other problem in the code. But I am still not sure why the closure is not executed while extending the DatabaseManager. I guess, that is the issue for some other day.
